src/docfx/lib/git/GitUtility.cs

```diff
@@ -70,6 +73,129 @@ public static unsafe List<GitCommit>[] GetCommits(string repoPath, List<string>
+        /// Pull update from remote
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <param name="remote">The remote name, default is origin</param>
+        /// <returns>Task status</returns>
+        public static Task Pull(string cwd, string remote = null)
+            => ExecuteNonQuery(cwd, $"pull {remote ?? string.Empty}", TimeSpan.FromMinutes(3));
+
+        /// <summary>
+        /// Checkout repo to specified branch
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <param name="create">Create this branch if not exists or not</param>
+        /// <param name="branch">The branch name, default is master</param>
+        /// <returns>Task status</returns>
+        public static Task Checkout(string cwd, bool create, string branch = null)
+            => ExecuteNonQuery(cwd, $"checkout {(create ? "-b" : "")} {branch ?? "master"}", TimeSpan.FromMinutes(5));
+
+        /// <summary>
+        /// Reset current repo to remote branch
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <param name="branch">The remote branch name</param>
+        /// <returns>Task status</returns>
+        public static Task Reset(string cwd, string branch)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(branch));
+
+            return ExecuteNonQuery(cwd, $"reset --hard origin/{branch}", TimeSpan.FromMinutes(10));
+        }
+
+        /// <summary>
+        /// Retrieve git head version
+        /// </summary>
+        /// <param name="cwd">The working directory</param>
+        /// <returns>The git head version</returns>
+        public static Task<string> HeadRevision(string cwd)
+           => ExecuteQuery(cwd, "rev-parse HEAD", TimeSpan.FromMinutes(3));
+
+        /// <summary>
+        /// Get commits (per file)
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <param name="file">The file path, can be null</param>
+        /// <param name="count">The commit count you want to retrieve, defualt is all</param>
+        /// <returns>A collection of git commit info</returns>
+        public static Task<IReadOnlyList<GitCommit>> GetCommits(string cwd, string file = null, int count = -1)
+        {
+            string formatter = "%H|%cI|%an|%ae|%cn|%ce";
+            var argumentsBuilder = new StringBuilder();
+            argumentsBuilder.Append($@"--no-pager log --format=""{formatter}""");
+            if (count > 0)
+            {
+                argumentsBuilder.Append($" -{count}");
+            }
+
+            if (!string.IsNullOrEmpty(file))
+            {
+                argumentsBuilder.Append($@" -- ""{file}""");
+            }
+
+            return ExecuteQuery(cwd, argumentsBuilder.ToString(), ParseListCommitOutPut);
+        }
+
+        private static IReadOnlyList<GitCommit> ParseListCommitOutPut(string lines)
```

Comment:
>OutPut [](start = 63, length = 6)

naming: `Output`

Reply:
ok

---
In reply to: [180617216](https://github.com/dotnet/docfx/pull/2611#discussion_r180617216) [](ancestors = 180617216)